src/pageObjects/signUpPage.js

```diff
@@ -0,0 +1,68 @@
+class SignUpPage {
+  constructor(page) {
+    this.page = page;
+    this.modalSelector = "app-signup-modal";
+    this.registerButtonSelector = "div.modal-footer > button";
+    this.errorSelector = ".alert.alert-danger";
+  }
+
+  async openSignUpModal() {
+    await this.page.goto("/");
+    await this.page.click("text=Sign Up");
+    await this.page.waitForSelector(this.modalSelector);
+  }
+
+  async signUpUser(signupData) {
+    for (const [input, value] of Object.entries(signupData)) {
+      await this.page.fill(`input#${input}`, value);
+    }
+    await this.clickRegisterButton();
+  }
+
+  async clickRegisterButton() {
+    const registerButton = await this.page.locator(this.registerButtonSelector);
+    const isButtonEnabled = await registerButton.isEnabled();
+    if (isButtonEnabled) {
+      await registerButton.click();
+    } else {
+      console.warn("Warning: Register button is disabled");
+    }
+  }
+
+  async waitForUserNavDropdown() {
+    await this.page.waitForSelector("#userNavDropdown");
+  }
+
+  async waitForExistingUserError() {
+    await this.page.waitForSelector(this.errorSelector);
+  }
+
+  async getExistingUserErrorText() {
+    return await this.page.textContent(this.errorSelector);
+  }
+
+  async waitForInputErrors(inputErrorSelectors) {
```

Comment:
each method should perform single action
read about single responsibility principle

For example
method `fill` accepts registration data and fills the form
method `waitForInputErrors` only perform assertions and checks input errors